src/main/plugin-metadata/plugin-security.policy

```diff
@@ -1,5 +1,8 @@
 grant {
     permission java.lang.RuntimePermission "loadLibrary.opensearchknn_nmslib";
     permission java.lang.RuntimePermission "loadLibrary.opensearchknn_faiss";
+    permission java.lang.RuntimePermission "loadLibrary.opensearchknn_faiss_avx2";
     permission java.net.SocketPermission "*", "connect,resolve";
+    permission java.lang.RuntimePermission "accessDeclaredMembers";
+    permission java.io.FilePermission "/proc/cpuinfo", "read";
```

Comment:
Will this cause any security issues?
Is there any lib that can do this for you instead of you reading through /proc/cpuinfo?

Reply:
> Will this cause any security issues? 

As a JNA maintainer, I'll note that the JNA core `ByReference` class uses reflection to ensure their subclasses implement the `byValue()` method.  For historical reasons, generic types weren't used so an abstract method wouldn't allow for the correct type to be used by implementing subclasses.  

This setting does invoke [some risk](https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimePermission.html):
> This grants code permission to query a class for its public, protected, default (package) access, and private fields and/or methods. Although the code would have access to the private and protected field and method names, it would not have access to the private/protected field data and would not be able to invoke any private methods. Nevertheless, malicious code may use this information to better aim an attack. Additionally, it may invoke any public methods and/or access public fields in the class. This could be dangerous if the code would normally not be able to invoke those methods and/or access the fields because it can't cast the object to the class/interface with those methods and fields.

There are some workarounds to avoid using the `size_t.ByReference` value in this code that I can help show how to use if Reflection is a concern.

> Is there any lib that can do this for you instead of you reading through /proc/cpuinfo?

As the maintainer/author of the OSHI library added in this PR, all it does is read through `/proc/cpuinfo` anyway.  But it's a library, so that makes it better. :) 

OSHI doesn't (yet) expose the flags as needed by this PR (although it does expose the udev "modalias" field which maps directly to flags and could help here).  But I plan on adding that capability soon (sooner if it's a blocker for this PR): https://github.com/oshi/oshi/issues/2589

However, note that `/proc/cpuinfo` isn't a normal file.  The whole `procfs` is a pseudo file system designed to expose internal kernel structures to users.  No file I/O occurs, and you get real-time data straight from the kernel.  So reading this "file" is really the most performant way to access that flag.

Reply:
Looks like it's used in several other plugins. Anything using an SPI needs it.

Reply:
yes, it is used in many plugins